packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { useRef, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import ThemeContext from 'terra-theme-context';
+import { FormattedMessage } from 'react-intl';
+import { KEY_SPACE, KEY_RETURN } from 'keycode-js';
+import { enableFocusStyles, removeFocusStyles, restoreFocusStyles } from './_SwitchUtils';
+import styles from './Switch.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const SWITCH_STATE = Object.freeze({
+  ON: <FormattedMessage id="Terra.switch.switchstatuslabel.on" />,
+  OFF: <FormattedMessage id="Terra.switch.switchstatuslabel.off" />,
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button.
+   */
+  switchId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  isChecked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled.
+   */
+  isDisabled: PropTypes.bool,
+  /**
+   * The label text of the Switch component.
+   */
+  labelText: PropTypes.string.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Returns Parameters: 1. switch value 2. event.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  isChecked: false,
+  isDisabled: false,
+  onChange: undefined,
+};
+
+const Switch = (props) => {
+  const {
+    switchId,
+    isChecked,
+    isDisabled,
+    onChange,
+    labelText,
+    ...customProps
+  } = props;
+  const theme = React.useContext(ThemeContext);
+  const sliderButton = useRef();
+
+  const handleOnClick = useCallback((event) => {
+    // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
+    // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
+    // This will set focus on the button when clicked.
+    sliderButton.current.focus();
+    if (onChange) {
+      onChange(!isChecked, event);
+    }
+  }, [isChecked, onChange]);
+
+  const handleOnKeyDown = (event) => {
+    if (event.nativeEvent.keyCode === KEY_RETURN || event.nativeEvent.keyCode === KEY_SPACE) {
+      event.preventDefault();
+      enableFocusStyles(sliderButton.current);
+      if (onChange) {
+        onChange(!isChecked, event);
+      }
+    }
+  };
+
+  const handleOnMouseDown = (event) => {
+    event.preventDefault();
```

Comment:
Remove prevent default on mouseDown.

Reply:
But it creates weird effects on space key down in IE like below hence added event.preventDefault(). 

![switch-IE-Dance1](https://user-images.githubusercontent.com/12869809/86386455-694a5800-bcaf-11ea-9274-f7d63de1469a.gif)

![switch-IE-Dance](https://user-images.githubusercontent.com/12869809/86386345-3d2ed700-bcaf-11ea-9197-73d4ead6c74a.gif)

Reply:
I tried removing this. but if we remove focus styles starts appear for mouse interactions in IE.

Reply:
Just checked, yeah you are correct, that part combined with the focus is correct. Disregard this comment.